bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/protocol/philipstv/PhilipsTVConnectionManager.java

```diff
@@ -0,0 +1,697 @@
+
+    public PhilipsTVConnectionManager(AndroidTVHandler handler, PhilipsTVConfiguration config) {
+
+        logger.debug("Create a Philips TV Handler for thing '{}'", handler.getThingUID());
+        this.handler = handler;
+        this.config = config;
+        this.scheduler = handler.getScheduler();
+        this.translationProvider = handler.getTranslationProvider();
+        this.discoveryServiceRegistry = handler.getDiscoveryServiceRegistry();
+        this.stateDescriptionProvider = handler.getStateDescriptionProvider();
+        this.target = new HttpHost(config.ipAddress, config.philipstvPort, HTTPS);
+        initialize();
+    }
+
+    private void setStatus(boolean isLoggedIn) {
+        if (isLoggedIn) {
+            setStatus(isLoggedIn, "online.online");
+        } else {
+            setStatus(isLoggedIn, "offline.unknown");
+        }
+    }
+
+    private void setStatus(boolean isLoggedIn, String statusMessage) {
+        String translatedMessage = translationProvider.getText(statusMessage);
+        logger.trace("setStatus to {} {} {}", isLoggedIn, statusMessage, translatedMessage);
+        if ((this.isLoggedIn != isLoggedIn) || (!this.statusMessage.equals(translatedMessage))) {
+            this.isLoggedIn = isLoggedIn;
+            this.statusMessage = translatedMessage;
+            handler.checkThingStatus();
+        }
+    }
+
+    public String getStatusMessage() {
+        return statusMessage;
+    }
+
+    public void setLoggedIn(boolean isLoggedIn) {
+        if (this.isLoggedIn != isLoggedIn) {
+            setStatus(isLoggedIn);
+        }
+    }
+
+    public boolean getLoggedIn() {
+        return isLoggedIn;
+    }
+
+    public void updateStatus(ThingStatus thingStatus, ThingStatusDetail thingStatusDetail, String thingStatusMessage) {
+        if (thingStatus == ThingStatus.ONLINE) {
+            setLoggedIn(true);
+        } else {
+            logger.trace("Updating status to {} {} {}", thingStatus, thingStatusDetail, thingStatusMessage);
+            setStatus(false, thingStatusMessage);
+        }
+    }
+
+    public String getMacAddress() {
+        return this.macAddress;
+    }
+
+    public void saveConfigs() {
+        String folderName = OpenHAB.getUserDataFolder() + "/androidtv";
+        File folder = new File(folderName);
+
+        if (!folder.exists()) {
+            logger.debug("Creating directory {}", folderName);
+            folder.mkdirs();
+        }
+
+        String fileName = folderName + "/philipstv." + handler.getThing().getUID().getId() + ".config";
+
+        Map<String, String> configMap = new HashMap<>();
+        configMap.put("username", username);
+        configMap.put("password", password);
+        configMap.put("macAddress", macAddress);
+
+        try {
+            String configJson = OBJECT_MAPPER.writeValueAsString(configMap);
+            logger.debug("Writing configJson \"{}\" to {}", configJson, fileName);
```

Comment:
My understanding is that the password here is a one time use between the user and the device during pairing.  This isn't a "semi-permanent" password to a cloud or anything.  Very similar to a pairing PIN that we use on the other protocols which we log.  I think the risk is very low since it's only valid for that short period of time.

Reply:
Thanks for the clarification.